### ISSUE TITLE
save the model promptly after training is done

### ISSUE THREAD

**Issue by alinelena** [ACEsuit/mace#847]
at the moment the sequence at the end of training/finetuning is
```
2025-03-06 11:13:09.611 INFO: Loaded Stage one model from epoch 9 for evaluation
2025-03-06 11:13:09.612 INFO: Evaluating train_pbe_d3 ...
2025-03-06 11:15:20.805 INFO: Evaluating train_pt_head ...
2025-03-06 12:24:15.590 INFO: Evaluating valid_pbe_d3 ...
2025-03-06 12:24:16.477 INFO: Evaluating valid_pt_head ...
2025-03-06 12:27:15.597 INFO: Error-table on TRAIN and VALID:
```
i wonder if makes sense to save the model before starting any evaluation...

**Comment by yixian929** [comment]
Based on my understanding, training is usually divided into training set, validation set, and test set. During the training process, the validation set loss, which is not used for training, is used to select the best batch model. The final evaluation mainly looks at the results of the test set.
please correct me if I am wrong

**Comment by alinelena** [comment]
I do not understand the comment in the context of this pr but here we go.

best model is evaluated on all sets (training validation [testing]) and a table is produced at the end, then the model is saved. in the case of fine tuning is the same only you have a more than one head... if for any reason you get an error in this processes you do not get a the mnodels saved.... this is what this pr tries to achieve. save best model before creating the tables and plotting
